doc/registration.md

```diff
@@ -0,0 +1,96 @@
+# Data registration queries
+
```

Comment:
The documentation starts goes directly into the meat of the matter (e.g. "These queries enable clients to find a single Registration by its unique ID."). 

Do you think it would be helpful to start with a short paragraph explaining what registrations are about?

Reply:
Thanks, added